### PR TITLE
added ability to specify protocol using CRN notation, along with a simple example notebook, script, and unit tests

### PR DESCRIPTION
**Pull request by dave-doty**

### CODE REVIEW

File: examples/test_crn.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        b + u >> 2 * b,
+    ]
+    n = 10 ** 5
+    init_config = {a: 0.51 * n, b: 0.49 * n}
+    sim = Simulation(init_config, approx_majority)
+    sim.run()
+    sim.history.plot()
+    plt.title('4 state majority protocol')
+    plt.xscale('symlog')
+    plt.yscale('symlog')
```
Comment (EricESeverson) [inline examples/test_crn.py]: plotting options from 4 state majority don't apply here

File: ppsim/crn.py [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+        products: Expression
+            The right hand side of the stoichiometric equation
+        k: float
+            The rate constant of the reaction
+    properties:
+        reactants: Expression
+            The left hand side of the stoichiometric equation
+        products: Expression
+            The right hand side of the stoichiometric equation
+        coeff: float
```
Comment (EricESeverson) [inline ppsim/crn.py]: Is this called coeff or rate_constant?

File: ppsim/crn.py [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+        Returns the net stoichiometric coefficient of a species in this
+        reaction.
+        args:
+            species: str
+                string name of the species
+        """
+        return (self.products.species.get(species_, 0) -
+                self.reactants.species.get(species_, 0))
+
+    def propensity(self, counts):
```
Comment (EricESeverson) [inline ppsim/crn.py]: What type is 'counts'? How is this expected to be used?

File: examples/test_crn.py (hunk 1 above)
Comment (dave-doty) [reply examples/test_crn.py]: fixed

File: ppsim/crn.py (hunk 2 above)
Comment (dave-doty) [reply ppsim/crn.py]: rate_constant

File: ppsim/crn.py (hunk 3 above)
Comment (dave-doty) [reply ppsim/crn.py]: This function has been removed.

**Review by EricESeverson — APPROVED** [review]

**Review by EricESeverson — APPROVED** [review]

**Pull request merged**